### PR TITLE
Fix geometry freeze  - MAGN 8958

### PR DESCRIPTION
**Pull request by ramramps**
## Purpose

The purpose of this PR is to add frozen state on geometry. As part of this PR,
1. alpha color component was changed to 0.5 for all the frozen nodes.
2. Nodes opacity is changed to 0.5 and will reflect only on the nodes. Preview bubble stays the same.
default

capture

This PR refers to the below tasks 
http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8958

## Design Document
 https://docs.google.com/a/adsk-oss.com/presentation/d/1MUhSzoDo-QFPysDv1E6gRslpq2pnb6eoPjKSsJN6g-o/edit?usp=sharing  

## Declaration
Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

## Reviewers
 @mjkkirschner

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@ namespace Dynamo.Wpf.ViewModels.Watch3D
     /// </summary>
     public static class AttachedProperties
     {
+        private const float transparentGeometry = 0.5f;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs]: can you rename this transparent geometry Multiplier

File: src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs [hunk 2]
```diff
@@ -63,5 +64,62 @@ public static bool GetHasTransparencyProperty(UIElement element)
...
+
+                for (int i = 0; i < colors.Length; i++)
+                {
+                    colors[i].Alpha = colors[i].Alpha * transparentGeometry;
+                }
+
+                geom.Geometry.Colors.Clear();
+                geom.Geometry.Colors.AddRange(colors);
+
+                var vertex = geom as DynamoGeometryModel3D;
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/ViewModels/Watch3D/AttachedProperties.cs]: this variable name doesn't seem to make sense to me, can you rename it or add a comment explaining why it should be called vertex?

**Comment by mjkkirschner** [comment]
@ramramps just a few small changes

**Comment by ramramps** [comment]
@mjkkirschner - PTAL.

**Comment by mjkkirschner** [comment]
@ramramps looks good to me

**Pull request merged**